lib/truffle/java/README.md

```diff
@@ -0,0 +1,89 @@
+# JRuby-compatible Java interop
+JRuby provides a rich set of interop functionality between Ruby and
+Java, and we attempt to emulate it faithfully in TruffleRuby.
+## Ruby proxies for Java classes
+In general a Ruby programmer using the interop API will never see raw
+Java objects, but instead either see primitive types which have been
+converted (numeric types and string), or other types wrapped in a
+proxy class. These classes are organised under the `::Java` module
+according to their Java package. For example `java.lang.Object` would
+have a proxy class created as `::Java::JavaUtil::Object`. To make it
+feel more natural when accessing Java classes they can also be
+accessed using `Kernel#java` to return the `java` package, with
+sub-package objects created as necessary, so you can write
+`java.lang.Object.new` to create a new instance of an object. In a
+similar fashion interfaces will have a corresponding module
+created. An interfaces module will be included in the proxy of any
```

Comment:
I did a very quick pass on this I'll admit since I already read it some time ago.